code/modules/projectiles/projectile/magic.dm

```diff
@@ -688,3 +690,276 @@
+		X.reagents.add_reagent(/datum/reagent/toxin, 10)
+
+/obj/item/projectile/magic/runic_death
+	name = "Runic Death"
+	icon_state = "antimagic"
+	flag = "magic"
+	impact_effect_type = /obj/effect/temp_visual/dir_setting/bloodsplatter
+
+/obj/item/projectile/magic/runic_death/on_hit(mob/living/target)
+	if(ismob(target))
+		var/mob/M = target
+		if(M.anti_magic_check())
+			M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			qdel(src)
+			return BULLET_ACT_BLOCK
+	. = ..()
+	if(iszombie(target))
+		target.gib()
+	if(isskeleton(target))
+		target.gib()
+	if(isvampire(target))
+		target.adjustBruteLoss(40)
+
+
+/obj/item/projectile/magic/shotgun_slug
+	name = "Shotgun slug"
+	icon_state = "bullet"
+	damage = 10
+	damage_type = BRUTE
+	nodamage = FALSE
+	flag = "magic"
+
+/obj/item/projectile/magic/shotgun_slug/on_hit(target)
+	if(ismob(target))
+		var/mob/M = target
+		if(M.anti_magic_check())
+			M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			qdel(src)
+			return BULLET_ACT_BLOCK
+	. = ..()
+
+/obj/item/projectile/magic/incediary_slug
+	name = "Incendiary shotgun slug"
+	icon_state = "bullet"
+	damage = 5
+	damage_type = BURN
+	nodamage = FALSE
+	flag = "magic"
+
+
+/obj/item/projectile/magic/incediary_slug/on_hit(target)
+	if(ismob(target))
+		var/mob/M = target
+		if(M.anti_magic_check())
+			M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			qdel(src)
+			return BULLET_ACT_BLOCK
+	. = ..()
+	if(iscarbon(target))
+		var/mob/living/carbon/X = target
+		X.fire_stacks += 1
+		X.IgniteMob()
+
+/obj/item/projectile/magic/runic_mutation
+	name = "Runic Mutation"
+	icon_state = "toxin"
+	flag = "magic"
+	irradiate = 12
+/obj/item/projectile/magic/runic_mutation/on_hit(target)
+	if(ismob(target))
+		var/mob/M = target
+		if(M.anti_magic_check())
+			M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			qdel(src)
+			return BULLET_ACT_BLOCK
+	. = ..()
+	if(iscarbon(target))
+		var/mob/living/carbon/X = target
+		X.randmuti()
+		if(prob(50))
+			X.easy_randmut(NEGATIVE)
+		else
+			if(prob(33))
+				X.easy_randmut(POSITIVE)
+			else
+				X.easy_randmut(MINOR_NEGATIVE)
+
+
+/obj/item/projectile/magic/runic_resizement
+	name = "Runic Resizement"
+	flag = "magic"
+	icon_state = "cursehand1"
+
+/obj/item/projectile/magic/runic_resizement/on_hit(target)
+	if(ismob(target))
+		var/mob/M = target
+		if(M.anti_magic_check())
+			M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			qdel(src)
+			return BULLET_ACT_BLOCK
+	. = ..()
+	if(isliving(target))
+		var/mob/living/X = target
+		var/newsize = pick(0.9, 0.8, 0.7, 0.6, 1.1, 1.2, 1.3, 1.4)
+		var/current_size = 1
+		X.resize = newsize*current_size
+		current_size = newsize
+		if(current_size < 0.25)	//if someone is smaller than 1/4 of their original size, it makes them not go that smol
```

Comment:
you could do something cheeky like having it just delete them for being too small, the chances of it are pretty slim as well